Vagrantfile

```diff
@@ -375,6 +376,18 @@ Vagrant.configure('2') do |config|
 
   # Configurations from 1.0.x can be placed in Vagrant 1.1.x specs like the following.
   config.vm.provider :virtualbox do |v|
+
+    unless Vagrant::Util::Platform.windows?
+      if Process.uid == 0
+        machine_id_file=Pathname.new(".vagrant/machines/default/virtualbox/id")
```

Comment:
What about other providers (VMware or parallels)?

Reply:
We don't know how to do this check for parallels or VMWare, or even if the `sudo` problem affects those users

Reply:
Should we do a check for the current provider then? And skip the warning? Or at least make it different?

Reply:
maybe is the case to do a check for the provider to avoid errors or crashes?

Reply:
I don't know that such a check would avoid errors and crashes, or what such a check would look like. I'm not even sure what the concern is?

Reply:
The problem is a design choice in VirtualBox (i.e.: it's intentional). Parallels doesn't have the same problem (though I can't swear it has _no_ problem if started as root, I only did a basic function test). Can't speak to VMWare.

Reply:
Actually, I'd argue that we should prevent using `sudo` with VirtualBox on all platforms. It's not _intended_ to be launched as root, so even if some OSs handle it better than others, it's still using it in a way other than intended. Even beyond that, I'd argue that we should at least discourage using `sudo` at all... there's just not really a good reason to do it, and it'll cause more issues than it solves 90% of the time.

Reply:
the problem is that this would mean using VVV on Windows would become extremely difficult as you couldn't launch it using an elevated prompt, which is necessary if you want hostnames and symlinks

Reply:
OK, so flip the conditions... make it always throw that error unless platform == windows and provider != virtualbox

Reply:
There's no *good* reason to use `sudo` on any platform but Windows... making Windows-based solutions other than VirtualBox the exception rather than the rule.

Reply:
`Unless` is used because the Codacy linter says it's better

Reply:
So? `Unless` can be applied either direction...

Reply:
That if not not windows

Reply:
Wait never mind, you're already catching it for non-windows systems elsewhere. You're good.

Reply:
Resolving this because:
 - the code is inside the virtualbox config section so other providers aren't affected
 - it doesn't run on Windows